airsonic-main/src/main/java/org/airsonic/player/dao/MediaFileDao.java

```diff
@@ -749,7 +749,7 @@ public void expunge() {
 
         final int batchSize = 1000;
         for (int id = minId; id <= maxId; id += batchSize) {
-            update("delete from media_file where id between ? and ? and not present", id, id + batchSize);
+            update("delete from media_file where id between ? and ? and (not present or folder not in (select path from music_folder))", id, id + batchSize);
```

Comment:
The revision policy is common to all Dao.
MusicFolder data that does not exist at the time of expunge is considered to be deleted.
(User data for which a dead record has already occurred can be rescued)

One thing to keep in mind is whether the present flag needs to be updated when the MusicFolder is deleted.

 - It has nothing to do with most existing processes, it should be....should ..
 - Before applying this code, you have to intentionally change the music folder and create a dirty record to verify. (In some cases, there may be cases in which we cannot help....?) Proof is difficult only with unit tests.